### PR TITLE
Implements the env delete command

### PR DESCRIPTION
**Pull request by efekarakus**
https://github.com/aws/amazon-ecs-cli-v2/issues/352 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/archer/env.go [hunk 1]
```diff
@@ -16,24 +16,30 @@ type Environment struct {
 	ManagerRoleARN string `json:"managerRoleARN"` // ARN for the manager role assumed to manipulate the environment and its applications.
 }
 
-// EnvironmentStore can List, Create and Get environments in an underlying project management store
+// EnvironmentStore can List, Create, Get, and Delete environments in an underlying project management store.
```
Comment (kohidave) [inline internal/pkg/archer/env.go]: 🎉

File: internal/pkg/cli/env_delete.go [hunk 2]
```diff
@@ -5,29 +5,129 @@ package cli
...
+
+	if opts.rgClient == nil {
+		// Tests mock the client.
+		if err := opts.initClientsFromRole(opts.env.ManagerRoleARN, opts.env.Region); err != nil {
+			return err
+		}
+	}
+
+	// Look up applications by searching for cloudformation stacks in the environment's region.
+	stacks, err := opts.rgClient.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
```
Comment (kohidave) [inline internal/pkg/cli/env_delete.go]: TODO - put this somewhere _else_. Maybe describe.

File: internal/pkg/cli/env_delete.go [hunk 3]
```diff
@@ -5,29 +5,129 @@ package cli
...
+	if err != nil {
+		return err
+	}
+
+	if !shouldDelete {
+		return nil
+	}
+	isStackDeleted := opts.deleteStack()
+	areRolesDeleted := opts.deleteRoles()
+	if isStackDeleted && areRolesDeleted {
```
Comment (kohidave) [inline internal/pkg/cli/env_delete.go]: We should eventually have a force command or something that will attempt to delete this parameter, even if we can't delete the stacks for some reason.

File: internal/pkg/cli/env_delete.go [hunk 4]
```diff
@@ -36,6 +136,83 @@ func (opts *DeleteEnvOpts) RecommendedActions() []string {
...
+	}
+	opts.prog.Stop(fmt.Sprintf(fmtDeleteEnvComplete, opts.env.Name, opts.env.Project))
+	return true
+}
+
+// deleteRoles returns true if the the execution role and manager role were deleted successfully. Otherwise, returns false.
+func (opts *DeleteEnvOpts) deleteRoles() bool {
+	// We must delete the EnvManagerRole last since as it's the assumed role for deletions.
+	roleNames := []string{
+		fmt.Sprintf("%s-%s-%s", opts.env.Project, opts.env.Name, "CFNExecutionRole"),
```
Comment (kohidave) [inline internal/pkg/cli/env_delete.go]: Future TODO: Figure out a better way of getting these... Maybe we can save it on the env it self. I think the ENV manager role is actually on the env object.

File: internal/pkg/deploy/cloudformation/stack/env.go [hunk 5]
```diff
@@ -120,3 +121,17 @@ func (e *EnvStackConfig) ToEnv(stack *cloudformation.Stack) (*archer.Environment
...
 	return &createdEnv, nil
 }
+
+// EnvStack is the cloudformation stack created for an environment.
+type EnvStack struct {
+	*cloudformation.Stack
+}
+
+// ExecutionRoleARN returns the execution role ARN from the stack.
+func (s *EnvStack) ExecutionRoleARN() string {
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/stack/env.go]: Maybe we could throw this into ToEnv ? That way you'd get the env manager role and execution role.

File: internal/pkg/cli/env_delete.go (hunk 4 above)
Comment (efekarakus) [reply internal/pkg/cli/env_delete.go]: Done!

File: internal/pkg/deploy/cloudformation/stack/env.go (hunk 5 above)
Comment (efekarakus) [reply internal/pkg/deploy/cloudformation/stack/env.go]: Done!

File: internal/pkg/cli/env_delete.go (hunk 3 above)
Comment (efekarakus) [reply internal/pkg/cli/env_delete.go]: Added TODO

File: internal/pkg/cli/env_delete.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/cli/env_delete.go]: Added TODO

**Pull request merged**